### PR TITLE
feat: (cherry-pick)(Version v12.2.0) Migration #122 set redesignedConfirmationsEnabled to true

### PR DESCRIPTION
**Pull request by digiwand**
## **Description**

Cherry-pick https://github.com/MetaMask/metamask-extension/pull/25769 for Version v12.2.0

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26043?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24614

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-background-state.json [hunk 1]
```diff
@@ -203,7 +203,8 @@
       "useNativeCurrencyAsPrimaryCurrency": true,
       "petnamesEnabled": true,
       "showTokenAutodetectModal": "boolean",
-      "isRedesignedConfirmationsDeveloperEnabled": "boolean"
+      "isRedesignedConfirmationsDeveloperEnabled": "boolean",
+      "redesignedConfirmationsEnabled": true
     },
     "ipfsGateway": "string",
     "isIpfsGatewayEnabled": "boolean",
```
Comment (digiwand) [inline test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-background-state.json:210]: snapshot differences:
1 less -1 line change
1 less +1 line change

CleanShot 2024-07-25 at 20 36 26

File: test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-ui-state.json [hunk 2]
```diff
@@ -255,6 +251,7 @@
       "swapsStxGetTransactionsRefreshTime": 10000,
       "swapsStxMaxFeeMultiplier": 2
     },
+    "bridgeState": { "bridgeFeatureFlags": { "extensionSupport": "boolean" } },
     "ensEntries": "object",
     "ensResolutionsByAddress": "object",
     "pendingApprovals": "object",
```
Comment (digiwand) [inline test/e2e/tests/metrics/state-snapshots/errors-after-init-opt-in-ui-state.json:257]: snapshot differences ([compare](https://github.com/MetaMask/metamask-extension/pull/25769/files#diff-da90b3618e8f510d2874ec229421985a86f88e8cf24a00a53a6563f5d56dc612L35)):
1 less - line change
1 less + line change

CleanShot 2024-07-25 at 20 38 20

File: app/scripts/migrations/index.js [hunk 3]
```diff
@@ -132,6 +132,7 @@ const migrations = [
   require('./119'),
   require('./120'),
   require('./121'),
+  require('./122'),
```
Comment (digiwand) [inline app/scripts/migrations/index.js:135]: 1 less -1 line change here ([compare](https://github.com/MetaMask/metamask-extension/pull/25769/files#diff-cd0f05c58fb43a4c8d16788608094f470253e904ebbd6f48d1894450f334e2c4L135))

**Review by digiwand — COMMENTED** [review]
line changes diff lgtm 👍🏼  

delta (-2 -3)

**Review by pnarayanaswamy — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]

**Review by hmalik88 — APPROVED** [review]

**Pull request merged**